hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java

```diff
@@ -164,11 +165,8 @@ public void handle(@NonNull final HandleContext handleContext) {
             builder.autoRenewAccountId(effectiveExpiryMeta.autoRenewAccountId());
 
             /* --- Add topic id to topic builder --- */
-            builder.topicId(TopicID.newBuilder()
-                    .shardNum(hederaConfig.shard())
-                    .realmNum(hederaConfig.realm())
-                    .topicNum(handleContext.entityNumGenerator().newEntityNum())
-                    .build());
+            builder.topicId(
+                    idFactory.newTopicId(handleContext.entityNumGenerator().newEntityNum()));
```

Comment:
Wouldn't it be cleaner if the `EntityNumGenerator` and `EntityIdFactory` combine so that you encapsulate both logic into `TopicId newTopicId()`?

Reply:
The `EntityNumGenerator` scope is per-dispatched transaction; while the `EntityIdFactory` is `@Singleton` scope.